### PR TITLE
Improve type handling in rich comparisons

### PR DESCRIPTION
**Pull request by PeterJCLaw**
Previously an error was raised when doing equality comparisons between BitMaps and other instance types, which is unexpected in Python. Move over to implementing each of the rich comparison methods directly so that we can more match the usual pattern of allowed (and rejected) comparisons.

Also explicitly check for `None` when validating that two BitMaps can be compared, so that the error message emitted in that case is of the more expected type (i.e: `TypeError` rather than `AttributeError`).

Fixes https://github.com/Ezibenroc/PyRoaringBitMap/issues/107 plus an issue with `None` found while writing the tests.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -400,12 +400,39 @@ def test_comparison(self, cls1, cls2, values1, values2, cow):
...
 
+    @given(bitmap_cls, hyp_collection, st.booleans())
+    def test_comparison_other_objects(self, cls, values, cow):
+        for op in [operator.le, operator.ge, operator.lt, operator.gt]:
+            bm = cls(values, copy_on_write=cow)
+            with self.assertRaises(TypeError):
+                op(bm, 42)
+            with self.assertRaises(TypeError):
+                op(bm, None)
+
```
Comment (PeterJCLaw) [inline test.py:411]: Hrm, should this be alongside `test_wrong_add` rather than here?

**Comment by PeterJCLaw** [comment]
Note: I suspect that this will conflict with #110; probably best to get that one in first as this PR will be the easier to re-work once that's in place.

**Review by Ezibenroc — APPROVED** [review]

**Comment by Ezibenroc** [comment]
> Note: I suspect that this will conflict with https://github.com/Ezibenroc/PyRoaringBitMap/pull/110; probably best to get that one in first as this PR will be the easier to re-work once that's in place.

Do you want to first merge #110 or #111 ?

**Comment by PeterJCLaw** [comment]
If we can merge #110 first I'm expecting that to be the easier way around.

**Comment by Ezibenroc** [comment]
Thank you !

**Pull request merged**